### PR TITLE
Bump upstream operator, serving and eventing to 0.21

### PR DESCRIPTION
**Pull request by matzew**
As per title:

* updated `go.mod` with latest version of 0.21 in first commit
* did the `make` target for proper updates afterwards

### CODE REVIEW

**Comment by matzew** [comment]
/hold
to make sure all really is fine and nice

File: go.mod [hunk 1]
```diff
@@ -32,13 +32,13 @@ require (
 	k8s.io/api v0.20.2
 	k8s.io/apimachinery v0.20.2
 	k8s.io/client-go v12.0.0+incompatible
-	knative.dev/eventing v0.20.1
+	knative.dev/eventing v0.21.3
```
Comment (markusthoemmes) [inline go.mod:35]: Please update the KNATIVE_VERSION in `./hack/update-deps.sh`. I'd think we should get https://github.com/openshift-knative/serverless-operator/pull/882 in first, then bump the version and run `./hack/update-deps.sh --upgrade` again for full consistency.

Comment (matzew) [reply go.mod:35]: Okay

**Comment by matzew** [comment]
rebasing, and followed the guideline from @markusthoemmes  - I forgot about that :smile_cat:

**Comment by matzew** [comment]
@nak3 is this some hint, that we should bump the serving bits too ?  :thinking:

**Comment by matzew** [comment]
perhaps we could to some of the bump from https://github.com/openshift-knative/serverless-operator/pull/879
(it's one commit, so hard to see what was updated before the generation)

**Comment by matzew** [comment]
@lberk FYI

for `eventing-kafka` we have the one JIRA for doing a proper automation. I am not yet touching the kafka bits here... perhaps it's possible to have that bump (and automation update) in a separate PR/bump.

**Comment by markusthoemmes** [comment]
/retest

Kourier images are now published.

**Comment by markusthoemmes** [comment]
More Ingress related breakage, fixing.

**Comment by markusthoemmes** [comment]
```
Unable to fetch image \"quay.io/openshift-knative/pizzaplanetv1:v0.21\": failed to resolve image to digest: HEAD https://quay.io/v2/openshift-knative/pizzaplanetv1/manifests/v0.21: unexpected status code 404 Not Found (HEAD responses have no body, use GET for details)."
```

Adding the relevant images for mirroring here: https://github.com/openshift/release/pull/17657

**Comment by matzew** [comment]
> Unable to fetch image \"quay.io/openshift-knative/pizzaplanetv1:v0.21\": failed to resolve image to digest: HEAD https://quay.io/v2/openshift-knative/pizzaplanetv1/manifests/v0.21: unexpected status code 404 Not Found (HEAD responses have no body, use GET for details)."

couldn't we also use the `IMAGE_TEMPLATE` stuff for the test images ?

**Comment by markusthoemmes** [comment]
/retest

Weird build issue.

**Review by markusthoemmes — COMMENTED** [review]
/lgtm
/approve

Abfahrt, as we say in Germany!

**Pull request merged**